src/components/limited-restoration-panel/LimitedRestorationPanel.vue

```diff
@@ -96,49 +95,38 @@ export default class LimitedRestorationPanel extends Vue {
   }
 
   /**
-   * Called when months input has changed.
+   * Called when radio value or input value have changed.
    */
-  async onMonthsInput (input: string): Promise<void> {
-    // ignore non-inputs
-    if (input !== null) {
-      // wait for component updates
+  @Watch('radioValue')
+  @Watch('inputValue')
+  private async onValueChanged (): Promise<void> {
+    if (this.radioValue === 'customMonths') {
+      // wait for component updates then validate the input field
       await this.$nextTick()
-      // validate the input
       const valid = this.$refs.monthsRef.validate()
-      // emit validity
+
+      // emit months and validity
+      this.emitMonths(valid ? +this.inputValue : null) // emit null if invalid
       this.emitValid(valid)
-      // if valid, emit new value
-      if (valid) this.emitMonths(Number(this.inputValue))
-    }
-  }
+    } else {
+      // reset input field validation
+      this.$refs.monthsRef.resetValidation()
 
-  /**
-   * Called when radio button has changed.
-   */
-  @Watch('radioValue')
-  private onRadioValueChanged () {
-    if (this.radioValue !== 'customMonths') {
-      // reset months text field when another radio button is selected
-      this.$refs.monthsRef.reset()
       // emit months and validity
-      this.emitMonths(Number(this.radioValue))
+      this.emitMonths(+this.radioValue)
```

Comment:
The [unary plus](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus) operator converts a string to a number more cleanup, imho.